File: 899. Orderly Queue/orderlyQueue.java
class Solution {
    public String orderlyQueue(String s, int k) {
        
        if(k == 1){
            //store
            String best = s;
            //check every possible SWITCH
            for(int i=0 ; i<s.length(); i++){
                String rev = s.substring(i) + s.substring(0,i);
                //compare
                if(rev.compareTo(best) < 0) best = rev;
            }
            return best;
        }else{
            //convert string to array of char
            char[] strs = s.toCharArray();
            //sort
            Arrays.sort(strs);
            //join
            return new String(strs);
        }
    }
}

File: 899. Orderly Queue/orderlyQueue.cpp
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    string orderlyQueue(string s, int k) {
        
        if(k == 1){
            string best = s;

            for(int i=1; i<best.size(); i++){
                string rev = s.substr(i) + s.substr(0,i);
                if(rev < best) best = rev;
            }
            return best;
        }else{
            string str = s;
            sort(str.begin(), str.end());
            return str;
        }
    }
};